src/main/java/seedu/task/model/util/DateParser.java

```diff
@@ -98,14 +98,27 @@ public static boolean isValidDate(int year, int month, int day, int hour, int mi
     private static int[] getDate(String date) throws IllegalValueException {
         Calendar calDate = Calendar.getInstance();
         int dayOfWeek = calDate.get(Calendar.DAY_OF_WEEK);
+        String firstWord;
+        if (date.indexOf(" ") == -1) {
+            firstWord = date;
+        } else {
+            firstWord = date.substring(0, date.indexOf(" "));
+        }
```

Comment:
Just FYI, [split](https://docs.oracle.com/javase/8/docs/api/java/lang/String.html#split-java.lang.String-) might be useful to you here